src/doc.d

```diff
@@ -768,19 +767,32 @@ extern (C++) static void emitAnchor(OutBuffer* buf, Dsymbol s, Scope* sc)
         emitAnchorName(&anc, s, skipNonQualScopes(sc));
         ident = Identifier.idPool(anc.peekString());
     }
-    size_t* count = cast(size_t*)dmd_aaGet(&sc.anchorCounts, cast(void*)ident);
-    TemplateDeclaration td = getEponymousParent(s);
-    // don't write an anchor for matching consecutive ditto symbols
-    if (*count > 0 && sc.prevAnchor == ident && sc.lastdc && (isDitto(s.comment) || (td && isDitto(td.comment))))
-        return;
-    (*count)++;
+
+    auto pcount = cast(void*)ident in sc.anchorCounts;
+    typeof(*pcount) count;
+    if (pcount)
+    {
+        // Existing anchor,
+        // don't write an anchor for matching consecutive ditto symbols
+        TemplateDeclaration td = getEponymousParent(s);
+        if (sc.prevAnchor == ident && sc.lastdc && (isDitto(s.comment) || (td && isDitto(td.comment))))
+            return;
+
+        count = ++*pcount;
+    }
+    else
+    {
+        sc.anchorCounts[cast(void*)ident] = 1;
```

Comment:
The cast is necessary because I don't want it to compare the class contents for identity, just the pointer value.